handbook/engineering/incidents/index.md

```diff
@@ -61,6 +64,24 @@ The owner of the incident may delegate tasks to other available/working engineer
 
 If the issue can not be quickly resolved (via rollback or other means) and if it is a severe problem with sourcegraph.com, then create an issue on sourcegraph/sourcegraph and tweet from the Sourcegraph account (e.g. https://twitter.com/srcgraph/status/1101603205203484672, https://twitter.com/srcgraph/status/1101606401753792512, https://twitter.com/srcgraph/status/1101621105620529153).
 
+
+## Escalation Policy
+
+When the resolution owner believes an incident requires more team members' attention, they may escalate the alert. By default, if an alert from our alerting infrastructure remains unacknowledged by the on-call engineer responsible for that alert for 10 minutes, then the entire mission team will be notified. The resolution owner may also trigger this manually within our alerting infrastructure. The resolution owner should make a best effort to contain an escalation to the team responsible for a service.
+
+Escalation Hierarchy:
+
+1. First one to discover issue / on-call engineer
+2. Mission team responsible for this service
+
+_most escalation should stop here_
+
+3. VP of Engineering
+   1. (Optional) If this a security incident then the security-team should be notified as well here
+4. CTO
+5. CEO
+
+
```

Comment:
I don't think we need a general escalation policy right now because the most important policy is encoded in how each team handles their oncall rotation.

Reply:
@pecigonzalo do you think its worth having a general escalation policy? I think updating this doc to the current state of team owned alerts is enough to satisfy the action item.

Reply:
I think what is missing is how to escalate beyond the team. For example, for the incident on Saturday, we were unable to find what was causing the failure and would have been to have a known escalation path. 
It was lucky that it was just `/pricing` and `/about` but if this was `sourcegraph.com`, it would have been a bigger issue.

In this example, I was unable to find were an IP was served from and needed to get someone online that might know were that was set, for example Stephen, Beyang, Quinn, Nick, as you some of these details are not documented.

The questions would have been:
- Who should I call in that case?
- How should I call them?
- Should I escalate for this?

I think we used to have a SWAT team, and I would propose we add back a similar thing and add those who might know those questions (or who to ask) in that group. It should not be linked to an alert, but rather manually triggered by an engineer if in this situation.

Reply:
I think you were still able to find the IP, though? I agree to have someone with more knowledge of this would have been helpful, but I'm not sure involving more people would have reduced the overall time to resolution. 

For your questions:
- Who should I have called? I believe notifying all on-call engineers was sufficient. 
- Via opsgenie ( `/genie whoisoncall` & `/genie alert "description of incident and link to Slack thread`) 
- I would argue if you have done both of the above you have escalated the issue to a reasonable degree. 

I would worry the SWAT team would become a permanent on-call team. That seems to go against the theme of this document which promotes containment of the unplanned work.

Reply:
> I think you were still able to find the IP, though?

We found the IP, we could not find what was serving the content behind the IP, it took us 3hours to find it in a comment in code.

> Via opsgenie ( /genie whoisoncall & /genie alert "description of incident and link to Slack thread)

That would work, but there is no group to notify other engineers that would have known this, notifying other engineering teams would have not made sense.

> I would argue if you have done both of the above you have escalated the issue to a reasonable degree.

There was no other escalation to reach, when I started troubleshooting the notification had already gone through the entire escalation for distribution.

>I would worry the SWAT team would become a permanent on-call team. That seems to go against the theme of this document which promotes containment of the unplanned work.

They would be the last escalation point, if they need to be pulled in constantly, then something is missing in our documentation or processes that on-call engineers are unable to fix the problems they are on-call for, as it was the case here. Each team is escalation for the person on-call if they cant answer, most alerts should not reach this point.
The problem is if this was sourcegraph.com I would have not known what to do. Im already the the last escalation of my rotation and I cant find a fix our source of the problem, should I have called someone? else and if so, who? 
The alternative is let the site be down until Monday, if that is acceptable, then we can document that.